salt/modules/win_lgpo.py

```diff
@@ -2828,13 +2863,14 @@ def _findOptionValueInSeceditFile(option):
     '''
     try:
         _d = uuid.uuid4().hex
-        _tfile = '{0}\\{1}'.format(__salt__['config.get']('cachedir'),
+        _tfile = '{0}\\{1}'.format(__opts__['cachedir'],
                                    'salt-secedit-dump-{0}.txt'.format(_d))
         _ret = __salt__['cmd.run']('secedit /export /cfg {0}'.format(_tfile))
         if _ret:
-            _reader = codecs.open(_tfile, 'r', encoding='utf-16')
-            _secdata = _reader.readlines()
-            _reader.close()
+            with io.open(_tfile, encoding='utf-16') as _reader:
+                _secdata = _reader.readlines()
+            while not _reader.closed:
+                _reader.close()
             if __salt__['file.file_exists'](_tfile):
                 _ret = __salt__['file.remove'](_tfile)
```

Comment:
@cachedout Could you look at the above code. On a few Windows boxes the old code here wasn't releasing the file handle before it tried to delete the file. This only happened on 5 machines out of 250 with one of our customers. What do you think of this approach? How best to avoid an infinite loop...?

Reply:
shouldn't the "with" ensure that it is cleaned up (since the original code did not use a "with")

wouldn't _reader be undefined outside of the "with" as well?

Reply:
You would think, but we tried this on the customer machine and the handle still was not released... adding a `time.sleep(1)` however, allowed enough time for the file handle to be released, but I don't really want to put a 1-second pause in every call to this function as it seems to get called a lot.

That's what I thought initially, but I still have access to `_reader` outside the with statement. I can return `_reader.name` which gives me the full path to the object and `_reader.closed` returns a boolean True/False. None of the calls to `_reader` are throwing an error, which would lead me to believe it's still accessible.